### PR TITLE
[IMP] web: Make loading extensible

### PR DESCRIPTION
**Pull request by yajo**
Description of the issue/feature this PR addresses:

Make loading changes through events. Mostly copied code from a single big unextensible method to small event-based methods, without changing current signature.

Current behavior before PR:

It becomes a nightmare if your addon needs to implement some specific behavior when loading or unloading. You either have to reimplement it, or rely on upstream implementation entirely.

Desired behavior after PR is merged:

With this patch, an event system is used to apply the loading change behaviors. This way, you can subscribe to them and react to those state changes.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr
@Tecnativa @ged-odoo

### CODE REVIEW

File: addons/web/static/src/js/chrome/loading.js [hunk 1]
```diff
@@ -41,36 +48,58 @@ var Loading = Widget.extend({
...
     on_rpc_event : function(increment) {
-        var self = this;
         if (!this.count && increment === 1) {
             // Block UI after 3s
-            this.long_running_timer = setTimeout(function () {
-                self.blocked_ui = true;
-                framework.blockUI();
-            }, 3000);
+            this.long_running_timer = setTimeout(
+                this.proxy("trigger_up"),
```
Comment (pedrobaeza) [inline addons/web/static/src/js/chrome/loading.js]: What is `proxy`? Can you point where is declared?

Comment (Yajo) [reply addons/web/static/src/js/chrome/loading.js]: https://github.com/odoo/odoo/blob/3b049e3395ebb53605b54cc352088ddecc753c9d/addons/web/static/src/js/core/mixins.js#L252-L280

Comment (pedrobaeza) [reply addons/web/static/src/js/chrome/loading.js]: We have found it, but please change it to current conventions:

```suggestion
                this.trigger_up.bind(this),
```

Comment (VincentSchippefilt) [reply addons/web/static/src/js/chrome/loading.js]: >       *

thanks, I wouldn't find it :)
It's a bit of magic that we don't use anymore

Comment (Yajo) [reply addons/web/static/src/js/chrome/loading.js]: Are you sure? That convention should be wrong as you can read in the docstring above.

Comment (pedrobaeza) [reply addons/web/static/src/js/chrome/loading.js]: Well, they are pretty sure that they don't use that since at least 2 years

Comment (VincentSchippefilt) [reply addons/web/static/src/js/chrome/loading.js]: Can you think of anything that would change the value of trigger_up between the time the setTimeout starts and the time setTimout actually executes the function ?

Comment (Yajo) [reply addons/web/static/src/js/chrome/loading.js]: If you don't use the proxy, then the method that's gonna be called is not gonna be extensible. You can use other alternatives if you prefer, but hard-binding is really not a good idea.

Modern JS equivalent would be: `() => this.trigger_up`

Comment (VincentSchippefilt) [reply addons/web/static/src/js/chrome/loading.js]: I'm good with () => this.trigger_up('loading.blocked')

**Comment by VincentSchippefilt** [comment]
@Yajo I like the idea of this change, could you meld your 2 commits together, I'll merge it when it's green

**Comment by Yajo** [comment]
It should be ready now.

**Comment by pedrobaeza** [comment]
@VincentSchippefilt I know now v13 is very settled, but maybe there's still a chance of getting this merged.

**Pull request closed without merging**